server/src/main/java/io/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -45,20 +46,60 @@
 
   @JsonCreator
   public CoordinatorDynamicConfig(
-      @JsonProperty("millisToWaitBeforeDeleting") long millisToWaitBeforeDeleting,
-      @JsonProperty("mergeBytesLimit") long mergeBytesLimit,
-      @JsonProperty("mergeSegmentsLimit") int mergeSegmentsLimit,
-      @JsonProperty("maxSegmentsToMove") int maxSegmentsToMove,
-      @JsonProperty("replicantLifetime") int replicantLifetime,
-      @JsonProperty("replicationThrottleLimit") int replicationThrottleLimit,
-      @JsonProperty("balancerComputeThreads") int balancerComputeThreads,
-      @JsonProperty("emitBalancingStats") boolean emitBalancingStats,
+      @JacksonInject JacksonConfigManager configManager,
+      @JsonProperty("millisToWaitBeforeDeleting") Long millisToWaitBeforeDeleting,
+      @JsonProperty("mergeBytesLimit") Long mergeBytesLimit,
+      @JsonProperty("mergeSegmentsLimit") Integer mergeSegmentsLimit,
+      @JsonProperty("maxSegmentsToMove") Integer maxSegmentsToMove,
+      @JsonProperty("replicantLifetime") Integer replicantLifetime,
+      @JsonProperty("replicationThrottleLimit") Integer replicationThrottleLimit,
+      @JsonProperty("balancerComputeThreads") Integer balancerComputeThreads,
+      @JsonProperty("emitBalancingStats") Boolean emitBalancingStats,
 
       // Type is Object here so that we can support both string and list as
       // coordinator console can not send array of strings in the update request.
       // See https://github.com/druid-io/druid/issues/3055
       @JsonProperty("killDataSourceWhitelist") Object killDataSourceWhitelist,
-      @JsonProperty("killAllDataSources") boolean killAllDataSources
+      @JsonProperty("killAllDataSources") Boolean killAllDataSources
+  )
+  {
+    CoordinatorDynamicConfig current = configManager.watch(
+        CoordinatorDynamicConfig.CONFIG_KEY,
+        CoordinatorDynamicConfig.class
+    ).get();
+    if (current == null) {
+      current = new Builder().build();
```

Comment:
not really, watch() returns a AtomicReference "watcher" which could return null at some point but a valid value later if config is modified in the database by anything.
we could add something configManager.get(T default) to get current value at some point but watch().get() is decent workaround and is used in most places.